File: teardown.rb
#!/usr/bin/env ruby

require 'yaml'

HOME = ENV['HOME']
LINKS = YAML::load_file 'config.yaml'

puts 'Tearing down...'
LINKS.each do |_, link|
  begin
    File.unlink File.join(HOME, link)
  rescue
    # ignore errors
  end
end
puts 'Done!'

File: teardown.hs
#!/usr/bin/env stack
-- stack --install-ghc runghc --package turtle
{-# LANGUAGE OverloadedStrings #-}

module Main where

import Turtle
import Prelude hiding (FilePath)
import qualified Data.Text as T

config :: FilePath
config = "config.yaml"

toLink :: Text -> FilePath
toLink = fromText . T.tail . T.dropWhile (/= ' ')

links :: Text -> [FilePath]
links = fmap toLink . T.lines

dotfiles :: IO FilePath
dotfiles = (\h -> h </> "github" </> "dotfiles") <$> home

unlink :: FilePath -> IO ()
unlink f = home >>= \h -> rm (h </> f)

main :: IO ()
main = do
  echo "Tearing down…"
  configText <- readTextFile config
  mapM_ unlink $ links configText
  echo "Done!"
